### ISSUE TITLE
Debianize Hailo

### ISSUE THREAD

**Issue by avar** [hinrik/hailo#25]
Let's make a Debian package for Hailo. That will allow us to be integrated into Debian / Ubuntu & take over the world.

**Comment by avar** [comment]
Running dh-make-perl --cpan Hailo gives:

```
Needs the following modules for which there are no debian packages available
 - Text::Trim
 - File::CountLines
 - IO::Interactive
 - MooseX::Role::Strict
= Test::More is a core module
= ExtUtils::MakeMaker is a core module
+ Test::Exception found in libtest-exception-perl
+ Data::Random found in libdata-random-perl
+ Test::Output found in libtest-output-perl
- Test::Exit not found in any package
   CPAN contains it in Test-Exit
   substituting package name of libtest-exit-perl
```

**Comment by avar** [comment]
Further tips:
- [Building Debian packages of Perl modules](http://www.debian-administration.org/articles/78)
- Use dh-make-perl --build --cpan Hailo
- lintian *deb gives some warnings:
  
  W: libhailo-perl: debian-copyright-file-uses-obsolete-national-encoding at line 2
  W: libhailo-perl: copyright-contains-dh-make-perl-boilerplate
  E: libhailo-perl: description-synopsis-is-duplicated
  W: libhailo-perl: description-contains-dh-make-perl-template
  W: libhailo-perl: maintainer-not-full-name avar
  E: libhailo-perl: perl-module-in-core-directory usr/lib/perl/5.10/
  W: libhailo-perl: package-installs-perllocal-pod usr/lib/perl/5.10/perllocal.pod
  E: libhailo-perl: perl-module-in-core-directory usr/lib/perl/5.10/perllocal.pod
  E: libhailo-perl: package-installs-packlist usr/lib/perl5/auto/Hailo/.packlist
  W: libhailo-perl: manpage-has-errors-from-man usr/share/man/man1/hailo.1p.gz 164: warning: macro `AE' not defined
  W: libhailo-perl: manpage-section-mismatch usr/share/man/man1/hailo.1p.gz:127 1p != 1
- Read HOW [TO GET A CPAN MODULE INTO DEBIAN](http://pkg-perl.alioth.debian.org/howto/RFP.html)
- It should be easy if we're willing to maintain it

**Comment by hinrik** [comment]
Yeah, but let's wait a bit before the DB format stabilizes and stuff.

**Comment by hinrik** [comment]
Looks like the revolution has already begun: http://www.mail-archive.com/debian-bugs-dist@lists.debian.org/msg745096.html

**Comment by avar** [comment]
A better link is here: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=568152

**Comment by mxey** [comment]
libfile-countlines-perl and libio-interactive-perl are currently in the NEW queue: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=568134 http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=568132

libtest-exit-perl is already in the pkg-perl Subversion repo, but unfished: 
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=568147. 

Text::Trim was missing a copyright statement. I have e-mailed the author, but have received no response so far. Without clear copyright assignment, this module cannot go into Debian. The ITP bug is http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=568144.

Feel free to help by taking on any of these problems, as I am not very active right now.

**Comment by avar** [comment]
I just removed our dependency on Text::Trim, we were only using it in one place anyway.

I'll make it so that Text::Exit is only optionally tested if it happens to be installed on the system, it's only a maintainer test anyway.

**Comment by avar** [comment]
Test::Exit is now fully optional.

Would it be easier if we maintained a debian/ directory in our repository?

**Comment by mxey** [comment]
no, we usually end up having to remove it from the tarball to keep our flexibility.

**Comment by mxey** [comment]
libtest-exit-perl is now in the NEW queue.

**Comment by avar** [comment]
Nice, although we don't need it anymore since it's conditionally part of the test suite.

Here's what dh-make-perl --build . gives me now:

```
Needs the following modules for which there are no debian packages available:
 - MouseX::Getopt
 - File::CountLines   
 - IO::Interactive
 - MouseX::Types
```

We switched Hailo by default to Mouse from Moose. It'll still work with just Moose but Mouse is recommended since it's faster & uses less memory. See [my blog posting](http://blogs.perl.org/users/aevar_arnfjor_bjarmason/2010/02/converting-a-moose-application-to-a-mousemoose-application.html) about this for more info.
Having MouseX::Getopt and MouseX::Types in Debian would be a good idea in any case. If not it'll still work just fine with Moose instead of Mouse.

**Comment by avar** [comment]
I don't know how Debian normally likes to package its packages but would it be possible to split Hailo up into a library package & command-line package? I.e.:
- hailo
- libperl-hailo
- libperl-poe-component-hailo
- libperl-poe-component-irc-plugin-hailo

The main motivation would be:
- A pretty name for the main program, like `meghal` but not `libmegahal++`
- Being able to ship a README + examples pertinent to either the library or the program

I'd also be interested in maintaining Debian packages for Hailo to take the load of the current Debian maintainers.

**Comment by fluctusz** [comment]
hi. any progress on this issue? as perl novice i miserably failed yesterday on installing hailo via cpanminus on a 64bit ubuntu10. any suggestions here?
thanks.

**Comment by avar** [comment]
We had a Debian ticked but there's been no response to it.
 On Oct 6, 2011 9:09 PM, "fluctusz" <
reply@reply.github.com>
wrote:

**Comment by fluctusz** [comment]
really hope this will work someday. thanks to keep on it.
regards, martin

**Comment by hinrik** [comment]
All the dependencies (except the optional Mouse ones) are in Debian now, but there hasn't been much activity about it in Debian's bug tracker. So now we just need some Debian developer to take interest in it and create the package.

Can you drop by in #hailo on irc.freenode.net or email me (hinrik.sig@gmail.com) about your issues with installing Hailo via cpanminus?
